src/main/java/org/wso2/extension/siddhi/window/uniquetimebatch/UniqueTimeBatchWindowProcessor.java

```diff
@@ -0,0 +1,335 @@
+    private long startTime = 0;
+    private VariableExpressionExecutor[] variableExpressionExecutors;
+
+    /**
+     * The setScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this setter method.
+     *
+     * @param scheduler the value of scheduler.
+     */
+    @Override
+    public void setScheduler(Scheduler scheduler) {
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * The getScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this getter method.
+     */
+    @Override
+    public Scheduler getScheduler() {
+        return scheduler;
+    }
+
+    /**
+     * The init method of the WindowProcessor, this method will be called before other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors,
+                        ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        variableExpressionExecutors = new VariableExpressionExecutor[attributeExpressionExecutors.length - 1];
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            variableExpressionExecutors[0] = (VariableExpressionExecutor) attributeExpressionExecutors[0];
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    timeInMilliSeconds = (Integer)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.LONG) {
+                    timeInMilliSeconds = (Long)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else {
+                    throw new ExecutionPlanValidationException("UniqueTimeBatch window's parameter time should be either" +
+                            " int or long, but found " + attributeExpressionExecutors[2].getReturnType());
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Time Batch window should have constant " +
+                        "for time parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            // isStartTimeEnabled used to set start time
+            isStartTimeEnabled = true;
+            if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                startTime = Integer.parseInt(String
```

Comment:
Please fix this. If you intend to support startTime for this Window, then this should be the third parameter. attributeExpressionExecutors[1] refers to windowSize.
If you're adding startTime as a parameter, then please modify init method accordingly and update documentation.

I think it's good to add startTime (as an optional parameter) because it's there in the TimeBatch Window as well.